src/Console/Command/GenerateExtensionCommand.php

```
@@ -10,116 +10,367 @@
     }
 
+    /**
+     * Returns option value otherwise ask user to input.
+     *
+     * @param string $question
+     * @param callable $checkValid
+     * @param string|null $default
+     * @return string
+     */
+    protected function getOptionOrAsk($option, $question, callable $checkValid, $default = null)
+    {
+        $input = $this->input->getOption($option);
+
+        if ($input) {
+            if (! $checkValid($input)) {
+                throw new Exception($option.' option is not valid');
+            }
+
+            return $input;
+        }
+
+        do {
+            $input = $this->ask($question, $default);
+        } while (! $checkValid($input));
+
+        return $input;
+    }
+
+    /**
+     * Ask user to input string.
+     *
+     * @param string $question
+     * @param string|null $default
+     * @return string
+     */
     protected function ask($question, $default = null)
     {
-        $question = new Question("<question>$question</question> ", $default);
+        $question = new Question(
+            '<question>'.$question.($default ? ' <comment>['.$default.']</comment>' : '').':</question> ',
+            $default
+        );
 
         return $this->getHelperSet()->get('question')->ask($this->input, $this->output, $question);
     }
 
+    /**
+     * Copy extension skeleton to destination.
+     *
+     * @param string $destination
+     * @param array $replacements
+     * @return void
+     */
     protected function copyStub($destination, $replacements = [])
     {
         $this->recursiveCopy(__DIR__.'/../../../stubs/extension', $destination, $replacements);
     }
 
+    /**
+     * Copy source to destination.
+     *
+     * @param string $src
+     * @param string $dst
+     * @param array $replacements
+     * @return void
+     */
     protected function recursiveCopy($src, $dst, $replacements = [])
     {
         $dir = opendir($src);
-        @mkdir($dst);
+        @mkdir($dst, 0775, true);
```

See my comment on #915 ; using `@` suppresses any error messages when creating a directory. The code should however check for its return code (true or false) whether it was succesful.